study-builder/src/main/java/org/broadinstitute/ddp/studybuilder/EmailBuilder.java

```diff
@@ -0,0 +1,213 @@
+            }
+        }
+        if (!emails.isEmpty()) {
+            Map<String, String> mappings = emails.stream().collect(Collectors.toMap(e -> e.key, e -> e.templateId));
+            LOG.info("Created {} email templates:\n{}", emails.size(), prettyGson.toJson(mappings));
+        }
+    }
+
+    private EmailInfo createSendGridEmail(Config emailCfg) {
+        String key = emailCfg.getString("key");
+        String templateId = ConfigUtil.getStrIfPresent(varsCfg, "emails." + key);
+        if (templateId != null && !templateId.isEmpty()) {
+            LOG.error("Email {} already has template id set to {}, not creating", key, templateId);
+            return null;
+        }
+
+        File file = cfgPath.getParent().resolve(emailCfg.getString("filepath")).toFile();
+        String html = readRawHtmlContent(file);
+        if (emailCfg.getBoolean("render")) {
+            html = renderHtmlContent(file.getName(), html);
+        }
+        String subject = emailCfg.getString("subject");
+        String name = emailCfg.getString("name");
+
+        var templateResult = sendGrid.createTemplate(name);
+        templateResult.rethrowIfThrown(e -> new DDPException("Error while creating email " + key, e));
+        if (templateResult.hasError()) {
+            LOG.error("Error while creating email {}: {}", key, templateResult.getError());
+            return null;
+        }
+
+        templateId = templateResult.getBody().getId();
+        String versionId = null;
+
+        var versionResult = sendGrid.createTemplateVersion(templateId, name, subject, html, true);
+        versionResult.rethrowIfThrown(e -> new DDPException("Error while creating version for email " + key, e));
+        if (versionResult.hasError()) {
+            LOG.error("Error while creating version for email {}. You might need to manually create"
+                            + " the version and run update-emails to load HTML content: {}",
+                    key, versionResult.getError());
+        } else {
+            versionId = versionResult.getBody().getId();
+        }
+
+        return new EmailInfo(key, templateId, versionId);
+    }
+
+    public void updateAll() {
+        updateForConfigs(List.copyOf(studyCfg.getConfigList("sendgridEmails")));
+    }
+
+    public void updateForEmailKeys(Set<String> keys) {
+        updateForConfigs(searchForConfigs(keys));
+    }
+
+    private void updateForConfigs(List<Config> emailCfgs) {
+        List<EmailInfo> emails = new ArrayList<>();
+        for (var emailCfg : emailCfgs) {
+            EmailInfo email = updateSendGridEmail(emailCfg);
+            if (email != null) {
+                emails.add(email);
+                LOG.info("Updated email {} with templateId={} versionId={}", email.key, email.templateId, email.versionId);
+            }
+        }
+        if (!emails.isEmpty()) {
+            LOG.info("Updated active versions of {} email templates", emails.size());
```

Comment:
Since the log entry has the size, I'd vote for remove the `if`.  Whoever is watching the logs would notice the `0` more than the lack of an entry?